src/features/user/BulkCodes/forms/IssueCodesForm.tsx

```diff
@@ -0,0 +1,275 @@
+
+const IssueCodesForm = ({}: IssueCodesFormProps): ReactElement | null => {
+  const { t, ready, i18n } = useTranslation(['common', 'bulkCodes']);
+  const router = useRouter();
+  const {
+    project,
+    setProject,
+    planetCashAccount,
+    projectList,
+    bulkMethod,
+    setBulkMethod,
+  } = useBulkCode();
+  const { user } = useContext(UserPropsContext);
+  const { getAccessTokenSilently } = useAuth0();
+  const { handleError } = useContext(ErrorHandlingContext);
+  const [localRecipients, setLocalRecipients] = useState<LocalRecipient[]>([]);
+  const [comment, setComment] = useState('');
+  const [occasion, setOccasion] = useState('');
+  const [codeQuantity, setCodeQuantity] = useState('');
+  const [unitsPerCode, setUnitsPerCode] = useState('');
+
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const resetBulkContext = (): void => {
+    setProject(null);
+    setBulkMethod(null);
+  };
+
+  const getProcessedRecipients = (): Recipient[] => {
+    const recipients: Recipient[] = [];
+    localRecipients.forEach((recipient) => {
+      const temp = {
+        recipientName: recipient.recipient_name,
+        recipientEmail: recipient.recipient_email,
+        message: recipient.recipient_message,
+        notifyRecipient: recipient.recipient_notify === 'yes',
+        units: parseInt(recipient.units),
+        // occasion: recipient.recipient_occasion,
+      };
+      recipients.push(temp);
+    });
+    return recipients;
+  };
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+    const token = await getAccessTokenSilently();
+    setIsProcessing(true);
+    if (project) {
+      const donationData = {
+        purpose: project.purpose,
+        project: project.guid,
+        prePaid: true,
+        comment,
+        treeCount: getTotalUnits(),
+        gift: {},
+      };
+      switch (bulkMethod) {
+        case BulkCodeMethods.GENERIC:
+          donationData.gift = {
+            type: 'code-bulk',
+            occasion,
+            numberOfCodes: Number(codeQuantity),
+            unitsPerCode: Number(unitsPerCode),
+          };
+          break;
+        case BulkCodeMethods.IMPORT:
+          donationData.gift = {
+            type: 'discrete-bulk',
+            occasion,
+            recipients: getProcessedRecipients(),
+          };
+          break;
+        default:
+          break;
+      }
+
+      const cleanedData = cleanObject(donationData);
+      const res = await postAuthenticatedRequest(
+        `/app/donations`,
+        cleanedData,
+        token,
+        handleError,
+        {
+          'IDEMPOTENCY-KEY': uuidv4(),
+        }
+      );
+      // if request is successful, it will have a uid
+      if (res.uid) {
+        resetBulkContext();
+        setIsSubmitted(true);
+        setTimeout(() => {
+          router.push(`/profile/history?ref=${res.uid}`);
+        }, 5000);
```

Comment:
Lets have a call with @jmiridis about the transaction taking time to appear in history.

Reply:
@Shreyaschorge This may work better on production, and it does not need to be addressed on priority. I've noted this down, will address if needed later on (as an update to release).

Reply:
Okay